biofefi/pages/feat_importance.py

```diff
@@ -1,192 +1,267 @@
+            ],
+            global_importance_methods=st.session_state[
+                ConfigStateKeys.GlobalFeatureImportanceMethods
+            ],
+        )
+    fi_opt = fi_opt.parse()
 
-st.session_state[ConfigStateKeys.EnsembleMethods] = ensemble_methods
+    return (
+        fuzzy_opt,
+        fi_opt,
+        st.session_state[ConfigStateKeys.ExperimentName],
+        st.session_state[ExplainModels.ExplainModels],
+    )
 
-st.write("### Local Feature Importance Methods")
-st.write(
-    "Select local methods to interpret individual predictions. "
-    "These methods focus on explaining predictions at a finer granularity."
-)
 
-local_importance_methods = {}
-use_lime = st.checkbox(
-    "LIME",
-    help="Use LIME (Local Interpretable Model-Agnostic Explanations) for local interpretability.",
-)
-local_importance_methods["LIME"] = {"type": "local", "value": use_lime}
-use_local_shap = st.checkbox(
-    "Local SHAP", help="Use SHAP for local feature importance at the instance level."
-)
-local_importance_methods["SHAP"] = {
-    "type": "local",
-    "value": use_local_shap,
-}
-
-st.session_state[ConfigStateKeys.LocalImportanceFeatures] = local_importance_methods
-
-st.write("### Additional Configuration Options")
-
-# Number of important features
-st.number_input(
-    "Number of most important features to plot",
-    min_value=1,
-    value=10,
-    help="Select how many top features to visualise based on their importance score.",
-    key=ConfigStateKeys.NumberOfImportantFeatures,
-)
+def save_upload(file_to_upload: str, content: str, mode: str = "w"):
+    """Save a file given to the UI to disk.
 
-# Scoring function for permutation importance
-st.selectbox(
-    "Scoring function for permutation importance",
-    [
-        "neg_mean_absolute_error",
-        "neg_root_mean_squared_error",
-        "accuracy",
-        "f1",
-    ],
-    help="Choose a scoring function to evaluate the model during permutation importance.",
-    key=ConfigStateKeys.ScoringFunction,
-)
+    Args:
+        file_to_upload (str): The name of the file to save.
+        content (str): The contents to save to the file.
+        mode (str): The mode to write the file. e.g. "w", "w+", "wb", etc.
+    """
+    base_dir = os.path.dirname(file_to_upload)
+    if not os.path.exists(base_dir):
+        os.makedirs(base_dir)
+    with open(file_to_upload, mode) as f:
+        f.write(content)
 
-# Number of repetitions for permutation importance
-st.number_input(
-    "Number of repetitions for permutation importance",
-    min_value=1,
-    value=5,
-    help="Specify the number of times to shuffle each feature for importance estimation.",
-    key=ConfigStateKeys.NumberOfRepetitions,
-)
 
-# Percentage of data to consider for SHAP
-st.slider(
-    "Percentage of data to consider for SHAP",
-    0,
-    100,
-    100,
-    help="Set the percentage of data used to calculate SHAP values.",
-    key=ConfigStateKeys.ShapDataPercentage,
-)
+def pipeline_fi(
```

Comment:
```suggestion
def pipeline(
```
This function is in the Feature Importance page, so it's redundant to add `_fi`